### PR TITLE
Added role consultant

### PR DESCRIPTION
**Pull request by Ashish944**
- Added role Consultant
- Added range for emp_ids of Consultant starting from 10K
- Added range for emp_ids of Bengaluru emps starting from 8K
- Consultant access - timesheet, documents, policy, profile, leave
- Removed Consultant emp from Website API
- Fix the timesheet index page
- Added leaves for Consultant role
- Changed the default path public profile for Consultant
- Restricted Access for Newsletter, Screamout and Employee list page
- Removed project list from Home page and timesheet forms
- Updated leave count for Consultant and removed remaining leave status
- Added data-table for manage leave template
- Updated task to reset yearly leave

### CODE REVIEW

File: app/models/concerns/leave_available.rb [hunk 1]
```diff
@@ -2,13 +2,13 @@ module LeaveAvailable
   extend ActiveSupport::Concern
   
   def assign_leave(event)
-    date_of_joining = self.private_profile.date_of_joining
+    doj = self.private_profile.date_of_joining
     self.employee_detail || self.build_employee_detail
     if (self.employee_detail.available_leaves == 0 || event == 'DOJ Updated') &&
        self.leave_applications.count == 0
-      self.employee_detail.available_leaves = calculate_leave(date_of_joining)
+      self.employee_detail.available_leaves = is_consultant? ? 150 : calculate_leave(doj)
```
Comment (Sameer-Tilak) [inline app/models/concerns/leave_available.rb:9]: Why 150 days of leaves assigned to consultant? Their each leave is deducted. Just as hack you can keep it at 150 for now, but once we add LWP type of leave their leave balance should always be 0.

File: app/models/user.rb [hunk 2]
```diff
@@ -314,12 +322,18 @@ def projects
...
+      employee_ids = employee_id_array.select{ |id| id > 10000 }
+      emp_id = employee_ids.empty? ? 10000 : employee_ids.max
+    elsif self.employee_detail.try(:location) == 'Bengaluru'
+      employee_ids = employee_id_array.select{ |id| id > 8000 && id < 9000 }
+      emp_id = employee_ids.empty? ? 8000 : employee_ids.max
+    elsif self.employee_detail.try(:location) == 'Plano'
+      usa_employee_ids = employee_id_array.select{ |id| id > 9000 && id < 10000}
       emp_id = usa_employee_ids.empty? ? 9000 : usa_employee_ids.max
     else
+      pune_employee_ids = employee_id_array.select { |id| id <= 9000}
```
Comment (Sameer-Tilak) [inline app/models/user.rb:336]: this condition should be id < 8000 and not 9000.

File: config/initializers/global_constants.rb [hunk 3]
```diff
@@ -20,7 +20,8 @@
 
 SLACK_API_TOKEN = ENV['SLACK_API_TOKEN']
 
-ROLE = { admin: 'Admin', employee: 'Employee', HR: 'HR', manager: 'manager', intern: 'Intern', team_member: 'team member' }
+ROLE = { admin: 'Admin', employee: 'Employee', HR: 'HR', manager: 'manager',
+         intern: 'Intern', team_member: 'team member', consultant: 'Consultant' }
```
Comment (Sameer-Tilak) [inline config/initializers/global_constants.rb:25]: We also have role as "Finance", Why it is missing in above list?

File: app/models/user.rb (hunk 2 above)
Comment (Ashish944) [reply app/models/user.rb:336]: Yes Sir, I'll update it.

File: app/models/concerns/leave_available.rb (hunk 1 above)
Comment (Ashish944) [reply app/models/concerns/leave_available.rb:9]: Yes Sir

**Comment by Sameer-Tilak** [comment]
LGTM

**Pull request merged**